framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -519,7 +509,6 @@ public T lockOneRandomRow(final SearchCriteria<T> sc, final boolean exclusive) {
         if (_removed != null) {
             sc.addAnd(_removed.second().field.getName(), SearchCriteria.Op.NULL);
```

Comment:
can use _checkAndSetRemovedIsNull()_ here ?

Reply:
Makes it a bit difficult since it is of a different type M not T

Reply:
ok @davidjumani